contracts/RewardPool.sol

```diff
@@ -150,7 +152,71 @@ contract RewardPool is Initializable, Versionable, Ownable {
       payableToken,
       amount
     );
-    return true;
+  }
+
+  function claim(
+    bytes calldata leaf,
+    bytes32[] calldata proof,
+    bool acceptPartialClaim
+  ) external nonReentrant returns (bool) {
+    (
+      address rewardProgramID,
+      ,
```

Comment:
Yeah this is worse to review / read IMO, sorry, I think library or previous approach if the library is too yak shavey.